site-src/geps/gep-1424.md

```diff
@@ -0,0 +1,381 @@
+
+`Service` resources supported as `backendRefs` SHOULD be consistent with expectations for North/South Gateway API implementations, MUST have associated endpoints, and include:
+
+* `ClusterIP` `Services` with selectors
+* [Headless Services](https://kubernetes.io/docs/concepts/services-networking/service/#headless-services) with selectors. Headless services without selectors SHOULD NOT be supported due to the possibility of matching `Services` with `type: ExternalName`.
+* `StatefulSet` resources
+
+#### `Service` without selectors
+
+An alternate pattern additionally supported by this approach would be to target a `Service` without selectors as the `parentRef`. This could be a clean way to create a pure routing construct and abstract a logical frontend, as traffic would resolve to a `backendRef` `Service` with selectors defined on the `HTTPRoute`, or receive a 4xx/5xx error response if no matching path or valid backend was found.
+
+
+#### Multicluster support with `ServiceImport`
+
+`ServiceImport` resources allocate a virtual IP in the cluster, so MAY be allowed as a `parentRef`.
+
+`ServiceImport` would remain a valid backend option for `HTTPRoute` resources (but _not_ currently a requirement for core conformance), and could be specified alongside a `Service` `backendRef` to split traffic across clusters within a `ClusterSet` (as defined in the [Multi-cluster Service (MCS) APIs project](https://github.com/kubernetes-sigs/mcs-api)). This could be a way to solve the need described in [A use case for using Gateway APIs in Multi-Cluster](https://docs.google.com/document/d/1bjr0uAVMmEtTX4mpU_aGXXapQFsEz_Qt0OnDoVswEEI/edit).
+
+### `hostnames` field
+
+GAMMA implementations SHOULD NOT infer any functionality from the `hostnames` field on `HTTPRoute` due to current under-specification and reserved potential for future usage or API changes. For the use case of filtering incoming traffic from selected hostnames, it is recommended to guide users toward configuring [`HTTPHeaderMatch`](https://gateway-api.sigs.k8s.io/v1alpha2/references/spec/#gateway.networking.k8s.io%2fv1beta1.HTTPHeaderMatch) rules for the [`Host`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Host) header. Functionality to be explored in future GEPs may include supporting concurrent usage of an `HTTPRoute` traffic configuration for multiple North/South `Gateways` and East/West mesh use cases or redirection of egress traffic to an in-cluster `Service`.
+
+### Limiting graph depth, preventing conflicts and cycles
+
+A `Service` may only be specified as a `parentRef` of an `HTTPRoute` if it is:
+
+* NOT already specified as a `parentRef` of another `HTTPRoute` [in the same namespace]
```

Comment:
Sorry, bit of a silly question but just wanted to make sure I understand this correctly. There can only be one route resource per `Service`?

Reply:
For initial implementation, yes - this will likely be relaxed in the future as described in the notes below because we know there is user interest in splitting large configurations over multiple files.